cmd/zoekt-git-clone/main.go

```diff
@@ -45,16 +47,22 @@ func main() {
 		log.Fatalf("url.Parse: %v", err)
 	}
 
-	name := filepath.Join(u.Host, u.Path)
-	name = strings.TrimSuffix(name, ".git")
+	name := *nameFlag
+	if name == "" {
+		name = filepath.Join(u.Host, u.Path)
+		name = strings.TrimSuffix(name, ".git")
+	}
+
+	repoId := *repoIdFlag
 
 	destDir := filepath.Dir(filepath.Join(*dest, name))
 	if err := os.MkdirAll(destDir, 0o755); err != nil {
 		log.Fatal(err)
 	}
 
 	config := map[string]string{
-		"zoekt.name": name,
+		"zoekt.name":   name,
+		"zoekt.repoid": repoId,
```

Comment:
I would only set this field if you have a non-empty ID. Additionally in other parts of Zoekt we assume the ID is a uint32. This is very sourcegraph specific and we can make it more generic if we want to make repoid a more first class citizen (which we do want in general)